<launch>

    <arg name="namespace" default="sam"/>
    <arg name="mode" default="default"/>
    <arg name="debug" default="0"/>
    <arg name="gps_port" value="/dev/ttyUSB2"/>
    <arg name="gps_baud" value="9600"/>
    <arg name="gps_frame_id" value="gps"/>

    <!-- Get these coordinates by clicking in Google maps and converting lat/lon to UTM -->
    <arg name="latitude" default="59.348513"/>
    <arg name="longitude" default="18.071553"/>
    <!-- UTM Zone: 34V -->
    <arg name="utm_easting" default="333492.08"/>
    <arg name="utm_northing" default="6582522.08"/>

    <node pkg="tf" type="static_transform_publisher" name="world_utm_broadcaster" args="$(arg utm_easting) $(arg utm_northing) 0 0 0 0 1 world_utm world_local 100" />
    <!-- This is required for the Aerial map display to work properly" -->
    <node pkg="tf" type="static_transform_publisher" name="map_utm_broadcaster" args="0 0 0 0 0 0 1 map world_utm 100" />

    <node pkg="rostopic" type="rostopic" name="fake_gps_fix"
          args="pub /fake_fix sensor_msgs/NavSatFix --latch
          '{header: {seq: 999, stamp: {secs: 0, nsecs: 0}, frame_id: 'world_local'},
            status: {status: 0, service: 1},
            latitude: $(arg latitude),
            longitude: $(arg longitude),
            altitude: 0,
            position_covariance: [3.9561210000000004, 0.0, 0.0, 0.0, 3.9561210000000004, 0.0, 0.0, 0.0, 7.650756],
            position_covariance_type: 2}'" output="screen"/>

    <node pkg="sam_dead_reckoning" type="publish_gps_path.py" name="publish_gps_path" output="screen"/>
    <node pkg="sam_dead_reckoning" type="spoof_gps_imu.py" name="spoof_gps_imu" output="screen"/>

    <!--
    <node pkg="nmea_navsat_driver" type="nmea_serial_driver" name="navsat" respawn="true">
        <param name="port" value="$(arg gps_port)"/>
        <param name="baud" value="$(arg gps_baud)"/>
        <param name="frame_id" value="$(arg gps_frame_id)"/>
    </node>
    -->

    <include file="$(find sam_dead_reckoning)/launch/sam_stim_dr.launch">
        <arg name="namespace" default="$(arg namespace)"/>
        <arg name="mode" default="$(arg mode)"/>
        <arg name="debug" default="$(arg debug)"/>
    </include>


</launch>
